src/lib/merkle_ledger/any_ledger.ml

```diff
@@ -14,16 +14,46 @@
  * *)
 
 open Core_kernel
-open Async_kernel
+
+module type S = sig
+  type key
+
+  type account
+
+  type hash
+
+  module Location : Location_intf.S
+
+  type witness [@@deriving sexp_of]
```

Comment:
why witness?

Reply:
It's a type that witnesses the base ledger signature

Reply:
ugh ok